src/asami/entities.cljc

```diff
@@ -174,20 +174,32 @@
                                (if-let [overflow (:overflow (ex-data e))]
                                  (reduced last-result)
                                  (throw e))))
-                           (if (and (seqable? obj)
+                           (if (and (seqable? obj)          ; would it be ok performance-wise to turn this into if-let with destructuring to extract e a v to use below instead of (nth ...)?
                                     (= 4 (count obj))
                                     (= :db/add (first obj)))
                              (or
-                              (when (= (nth obj 2) :db/id)
+                               ;; Ex.: `[:db/add [:id X] :id X]` that creates a new entity
+                               (when-let [ref (and (writer/lookup-ref? (nth obj 1))
+                                                   (= (first (nth obj 1)) (nth obj 2))
+                                                   (nth obj 1))]
+                                 (let [new-id (or #_(ids ref) (node/new-node graph))] ; I cannot see where `(ids ref)` would be non-nil in any well-formed tx
+                                   [(conj acc (assoc (vec-rest obj) 0 new-id))
+                                    racc
+                                    (assoc ids ref new-id)
```

Comment:
This will add something like `[:id "bobid"] :a/node-28618` into the `ids` map => it will then work for lookups in the `(ids %)` calls at line 199 BUT it is inconsistent with what `entity-triples` does, which - given an input such as `{:id "maryid", ..}` will add to `ids` something like `"maryid" :a/node-28619`. What is right, I do not know. But I would think that the original code, in entity-triples, is more right and thus we need to change what we put in ids and how we do the lookup.

Reply:
/holyjak/asami#2 addresses ☝️ and is intended to start a discussion about what is the right approach here.